agreement/actions.go

```diff
@@ -314,8 +340,22 @@ func (a rezeroAction) String() string {
 	return a.t().String()
 }
 
+func (a rezeroAction) ComparableString() string {
+	return fmt.Sprintf("%s: %d", a.t().String(), a.Round)
+}
+
 func (a rezeroAction) do(ctx context.Context, s *Service) {
 	s.Clock = s.Clock.Zero()
+	if _, ok := s.historicalClocks[a.Round]; !ok {
+		s.historicalClocks[a.Round] = s.Clock
+	}
```

Comment:
If and old round can comes in, why do we Zero the Service's main clock, `s.Clock`? Maybe that ends up unused in such cases?

I think I want to advocate getting rid of `s.Clock` and renaming `s.historicClocks` to `s.RoundStarts`, which should also have the clock that would be `s.Clock` today inside of it. That would simplify `getTimstampForEvent` which is currently a bit odd. The second argument comes in, already relative to `s.Clock`, but that argument is sometimes ignored, and a new call to Since() is made against the appropriate clock from `historicalClocks`. Seems like it could just always be a `Since()` from the appropriate Clock in `roundStarts`.

Reply:
But this _does_ reset the main clock, even processing an old event, doesn't it?

Reply:
Can you do it the other way around, stop passing historicalClocks at all, and instead pick the single correct Clock to pass in from the caller?

Reply:
I guess that precludes the basics.Duration(0) case, which I don't understand.

Reply:
`func clockForRound()` seems like an easy to understand idea, and then you just pass the right clock in, without the round.

But I suppose you'd need something to return for the future round case that will return 1 for `Since()`

Reply:
I don't want to bike shed you any more than I already have, so I'm ok with any way you want to pass the info.

But I do want to understand why it's ok to `s.Clock = s.Clock.Zero()` in the first line, when it may still turn out that you want to adjust an old clock, not `s.Clock`.